core/consensus/authority/impl/authority_manager_impl.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CONSENSUS_AUTHORITIES_MANAGER_IMPL
+#define KAGOME_CONSENSUS_AUTHORITIES_MANAGER_IMPL
+
+#include <primitives/babe_configuration.hpp>
+#include "consensus/authority/authority_manager.hpp"
+#include "consensus/authority/authority_update_observer.hpp"
+#include "consensus/grandpa/finalization_observer.hpp"
+
+#include "application/app_state_manager.hpp"
+#include "blockchain/block_tree.hpp"
+#include "consensus/authority/impl/schedule_node.hpp"
+#include "storage/buffer_map_types.hpp"
+
+namespace kagome::authority {
+  class AuthorityManagerImpl : public AuthorityManager,
+                               public AuthorityUpdateObserver,
+                               public consensus::grandpa::FinalizationObserver {
+   public:
+    inline static const std::vector<primitives::ConsensusEngineId>
+        known_engines{primitives::kBabeEngineId, primitives::kGrandpaEngineId};
+    inline static const common::Buffer SCHEDULER_TREE =
+        common::Buffer{}.put(":kagome:authorities:scheduler_tree");
+
+    AuthorityManagerImpl(
+        std::shared_ptr<application::AppStateManager> app_state_manager,
+        std::shared_ptr<primitives::BabeConfiguration> genesis_configuration,
+        std::shared_ptr<blockchain::BlockTree> block_tree,
+        std::shared_ptr<storage::BufferStorage> storage);
+
+    ~AuthorityManagerImpl() override = default;
+
+    void prepare();
```

Comment:
It's one of method what is using AppStateManager. I'm not sure we need to comment this everywhere.

Reply:
Well, since I don't see an interface or something which would clearly send me to AppStateManager here, I think that at least something like /** @see AppStateManager */ would do.

Reply:
Wiil be done